### PR TITLE
Base: Implement receiver action

### PR DESCRIPTION
**Pull request by facuspagnuolo**
Receiver actions allow users to send any assets to them to trigger specific actions.
This implementation is simplifies this model by making the receiver an action as well.

Features covered:
- Receiving assets
- Allow forwarding assets to the SV
- Allow collecting assets from the SV to use it from an action

### CODE REVIEW

File: packages/base/contracts/actions/ReceiverAction.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    function withdraw(address token, uint256 amount) external auth {
+        require(amount >= _balanceOf(token), 'RECEIVE_INSUFFICIENT_BALANCE');
+        Denominations.isNativeToken(token)
+            ? Address.sendValue(payable(address(smartVault)), amount)
+            : IERC20(token).safeTransfer(address(smartVault), amount);
+    }
+
+    function _collect(address token, uint256 amount) internal {
```
Comment (dmf7z) [inline packages/base/contracts/actions/ReceiverAction.sol:39]: What is the idea behind to have a collect and a withdraw function?

Comment (facuspagnuolo) [reply packages/base/contracts/actions/ReceiverAction.sol:39]: In a certain way these model the same idea, but I thought `withdraw` as an emergency exit, something that the owner is allowed to call to send assets to the SV without any further logic like wrapping, approving, collecting, etc... What do you think?

Comment (dmf7z) [reply packages/base/contracts/actions/ReceiverAction.sol:39]: ok, I was thinking they can do a collet and witdraw from the SV, but this is more direct, all right

**Review by dmf7z — APPROVED** [review]

File: packages/base/contracts/actions/ReceiverAction.sol (hunk 1 above)
Comment (dmf7z) [reply packages/base/contracts/actions/ReceiverAction.sol:39]: I was also thinking that maybe we can remove the collect and use an internal  _withdraw when executing the action

**Review by dmf7z — APPROVED** [review]

**Pull request merged**